src/sql/views.d.ts

```diff
@@ -0,0 +1,27 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'views' {
+	import * as vscode from 'vscode';
+	import * as data from 'data';
+
+	export interface Button {
+		label: string;
+		readonly onClicked: vscode.Event<any>;
+	}
+
+	export interface UIControl {
+		type: data.ControlTypes,
+		control: any,
+		container: string,
+		id: number
+	}
+
+	export interface ControlEventArgs {
+		type: data.ControlTypes,
+		id: number,
+		event: string
+	}
+}
```

Comment:
Could we move this into the existing `data` api under a "views" namespace? I think `data` is designed to be SQL Ops Studio's top-level extensibility API, just like `vscode` for VS Code.

Reply:
I feel that there are too many interfaces already there and for UI stuff I would move them all to a new file. I wanted to move the dialog to this file also 

---
In reply to: [166082426](https://github.com/Microsoft/sqlopsstudio/pull/603#discussion_r166082426) [](ancestors = 166082426)